### PR TITLE
refactor: Bootstrap to AntD - DropdownButton

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
- Migrates Collapse component from Bootstrap to AntD.
- Aligns dropdown caret with text

See: #10254

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screen Shot 2021-02-05 at 2 50 34 PM
Screen Shot 2021-02-05 at 2 50 34 PM
Screen Shot 2021-02-05 at 2 51 08 PM
Screen Shot 2021-02-08 at 10 22 04 AM
Screen Shot 2021-02-08 at 10 20 29 AM
Screen Shot 2021-02-08 at 10 21 16 AM
Screen Shot 2021-02-08 at 10 52 39 AM
Screen Shot 2021-02-08 at 10 53 11 AM

@rusackas @junlincc

### TEST PLAN
1 - Open any screen that contains a `DropdownButton` (explore, dashboard -> edit, etc.)
2 - All `DropdownButton` components should have the same theme and behavior

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by junlincc — COMMENTED** [review]
approving as product sign-off. ✅ thanks for the PR! 
now i know where the previous mysterious drop-shadow in Explore came from lol 
Screen Shot 2021-02-08 at 10 24 08 PM
Screen Shot 2021-02-08 at 10 24 53 PM

**Comment by junlincc** [comment]
not related, @michael-s-molina Michael, could you double check if we covered the casing change in the main menu dropdown? thanks! 
Screen Shot 2021-02-08 at 10 29 28 PM

**Comment by michael-s-molina** [comment]
> not related, @michael-s-molina Michael, could you double check if we covered the casing change in the main menu dropdown? thanks!
> Screen Shot 2021-02-08 at 10 29 28 PM

@junlincc This text is translated on the server side (Python). This is the part that Kasia was working on. We should discuss how to continue this today in our meeting.

File: superset-frontend/src/dashboard/components/menu/PopoverDropdown.jsx [hunk 1]
```diff
@@ -89,25 +90,31 @@ class PopoverDropdown extends React.PureComponent {
...
     const selected = options.find(opt => opt.value === value);
     return (
-      <DropdownButton
+      <Dropdown
         id={id}
-        bsSize="small"
-        title={renderButton(selected)}
-        className="popover-dropdown"
+        trigger="click"
+        overlayStyle={{ zIndex: 3001 }}
```
Comment (rusackas) [inline superset-frontend/src/dashboard/components/menu/PopoverDropdown.jsx:96]: I'm a little hesitant about random zIndex values re-appearing in the codebase after making some effort to consolidate them. I see a couple paths forward:
• Making a z-index registry in the Theme (painful for now, due to the external repo)
• Exporting a z-index value from the Popover itself, and importing here to reference/increment.

File: superset-frontend/src/dashboard/components/menu/PopoverDropdown.jsx [hunk 2]
```diff
@@ -78,6 +78,7 @@ const MenuItem = styled(Menu.Item)`
 class PopoverDropdown extends React.PureComponent {
```
Comment (rusackas) [inline superset-frontend/src/dashboard/components/menu/PopoverDropdown.jsx:78]: Maybe we should add a Storybook entry? Could be a separate PR, or we can do a whole audit of missing storybook entries after we close the books on some of the current audit/transition efforts.

File: superset-frontend/src/dashboard/components/menu/PopoverDropdown.jsx [hunk 3]
```diff
@@ -89,25 +90,31 @@ class PopoverDropdown extends React.PureComponent {
...
-              })}
-            >
-              {renderOption(option)}
-            </MenuItem>
-          ))}
-        </Menu>
-      </DropdownButton>
+        <div role="button" css={{ display: 'flex', alignItems: 'center' }}>
+          {renderButton(selected)}
+          <Icon name="caret-down" css={{ marginTop: 4 }} />
```
Comment (rusackas) [inline superset-frontend/src/dashboard/components/menu/PopoverDropdown.jsx:115]: Not sure if this measurement makes more sense based on gridUnit. ¯\\\_(ツ)_/¯

File: superset-frontend/src/dashboard/components/menu/WithPopoverMenu.jsx [hunk 4]
```diff
@@ -38,7 +38,9 @@ const defaultProps = {
   menuItems: [],
   isFocused: false,
   shouldFocus: (event, container) =>
-    container && container.contains(event.target),
+    (container && container.contains(event.target)) ||
```
Comment (rusackas) [inline superset-frontend/src/dashboard/components/menu/WithPopoverMenu.jsx:41]: Could use the null coalescing operator here, 
```suggestion
    (container?.contains(event.target)) ||
```

File: superset-frontend/src/dashboard/components/menu/WithPopoverMenu.jsx [hunk 5]
```diff
@@ -38,7 +38,9 @@ const defaultProps = {
...
+    event.target.id === 'menu-item' ||
+    (event.target.parentNode && event.target.parentNode.id === 'menu-item'),
```
Comment (rusackas) [inline superset-frontend/src/dashboard/components/menu/WithPopoverMenu.jsx:43]:
```suggestion
    (event.target.parentNode && event.target.parentNode?.id === 'menu-item'),
```
... and here

File: superset-frontend/src/explore/components/DisplayQueryButton.jsx [hunk 6]
```diff
@@ -156,48 +153,47 @@ export const DisplayQueryButton = props => {
...
-    </DropdownButton>
+        </Menu>
+      }
+    >
+      <div
+        role="button"
+        id="query"
+        tabIndex={0}
+        className="btn btn-default btn-sm"
+      >
```
Comment (rusackas) [inline superset-frontend/src/explore/components/DisplayQueryButton.jsx:193]: Maybe we can replace this with the shiny new Button component?

**Review by rusackas — APPROVED** [review]
A few nits/comments, and places we can expand, but in general this approach looks great to me! Thanks for your patience with this. @junlincc (and @eschutho?) let's find some time to QA-bash this and get it in!

File: superset-frontend/src/dashboard/components/menu/PopoverDropdown.jsx (hunk 1 above)
Comment (michael-s-molina) [reply superset-frontend/src/dashboard/components/menu/PopoverDropdown.jsx]: I changed to pull `z-index` from the theme for now but I really like the idea of a z-index registry. We should do that in the future.

File: superset-frontend/src/dashboard/components/menu/PopoverDropdown.jsx (hunk 2 above)
Comment (michael-s-molina) [reply superset-frontend/src/dashboard/components/menu/PopoverDropdown.jsx:78]: I agree with the suggestion in a separate PR. I added this component to our tests audit spreadsheet with the following notes: 

> We should move this to the components folder, change it to a functional component with typescript, create tests and storybook

File: superset-frontend/src/dashboard/components/menu/PopoverDropdown.jsx (hunk 3 above)
Comment (michael-s-molina) [reply superset-frontend/src/dashboard/components/menu/PopoverDropdown.jsx:122]: Not this one 😆. 

The funny thing is that every time I encounter a multiple of 4 your image comes to my mind with the words "**grid unit**" 🤣

File: superset-frontend/src/dashboard/components/menu/WithPopoverMenu.jsx (hunk 4 above)
Comment (michael-s-molina) [reply superset-frontend/src/dashboard/components/menu/WithPopoverMenu.jsx]: Great suggestion! Done.

File: superset-frontend/src/dashboard/components/menu/WithPopoverMenu.jsx (hunk 5 above)
Comment (michael-s-molina) [reply superset-frontend/src/dashboard/components/menu/WithPopoverMenu.jsx]: Great suggestion! Done.

File: superset-frontend/src/explore/components/DisplayQueryButton.jsx (hunk 6 above)
Comment (michael-s-molina) [reply superset-frontend/src/explore/components/DisplayQueryButton.jsx:193]: Agree. Separate PR.

File: superset-frontend/src/dashboard/components/menu/PopoverDropdown.jsx (hunk 3 above)
Comment (rusackas) [reply superset-frontend/src/dashboard/components/menu/PopoverDropdown.jsx:122]: I know... it's an annoying pedantic thing I do on every PR. It's always "use the theme colors" or "use gridUnit" everywhere I go. Sorry to be _that guy_.

Comment (michael-s-molina) [reply superset-frontend/src/dashboard/components/menu/PopoverDropdown.jsx:122]: I really prefer this detailed review to a superficial one. It's very important to keep our standards intact. 😉
